### ISSUE TITLE
Transferring reverse encfs SMB/AFP share fails when accessing large folders

### ISSUE THREAD

**Issue by gryphonmyers** [ncw/rclone#1008]
rclone v1.35
macOS Sierra 64bit
ACD

/usr/local/bin/rclone sync --bwlimit 400 --log-file=/Users/gryphon/Desktop/log.txt -v /Users/gryphon/Encrypted/2TDYZCWWFS3LTLST7OPUBM542ALIL/ remote:/Encrypted/2TDYZCWWFS3LTLST7OPUBM542ALIL/

https://www.dropbox.com/s/dgzsc622smm13cp/log.txt?dl=0

I am encountering periodic, seemingly random errors when I attempt to upload my reverse encfs view of an AFP mount. The error I always get is something resembling: 

2017/01/09 16:05:53 Attempt 3/3 failed with 0 errors and: error listing source: Local file system at /Users/gryphon/Encrypted/2TDYZCWWFS3LTLST7OPUBM542ALIL: failed to read directory "/Users/gryphon/Encrypted/2TDYZCWWFS3LTLST7OPUBM542ALIL/TLYP4YMX36FKWV2SRZPIOWC3EWGWA/MUL7FSSGJW52R4XXNYD6S6L6NKXWH/RBVOB5PVSHODPPNGIRYYD5VO55LHKSH4BA5QU44CWPPIY3MYIER54P2KGJ6OP6JNJDT2DGVEPRLDXR5U": readdirent: input/output error

Every time I encounter this error, I am able to cd to the resource it complains about with no problems. I am wondering if it is timing out on directories containing lots of files, as since it is mounted over the network very large folders might experience latency. I didn't see any relevant settings on the sync command. 

I'd appreciate any help tracking the cause of this issue down!

**Comment by friendlytherapist** [comment]
When you enter the same command, sometimes it will error and sometimes it will go through the upload fine?

**Comment by gryphonmyers** [comment]
If I immediately run the same command, it does appear to get hung up on the same folder almost immediately. But if I restart then try again, it won't necessarily error on the file (possibly because it starts off at a different index and doesn't get to it?) It's just confusing because like I said I can access the folder it's complaining about (~/Encrypted/2TDYZCWWFS3LTLST7OPUBM542ALIL/TLYP4YMX36FKWV2SRZPIOWC3EWGWA/MUL7FSSGJW52R4XXNYD6S6L6NKXWH/RBVOB5PVSHODPPNGIRYYD5VO55LHKSH4BA5QU44CWPPIY3MYIER54P2KGJ6OP6JNJDT2DGVEPRLDXR5U) just fine in terminal and in finder. I did notice though, that if I navigate to MUL7FSSGJW52R4XXNYD6S6L6NKXWH (the parent of the folder it's complaining about) there was maybe a 5-10 second wait while it loaded all the files. This is why I wonder if some timeout threshold could be causing it.

**Comment by ncw** [comment]
The error `readdirent: input/output error` is encfs returning EIO while reading a directory.  Can you run encfs with more logging and have a look at its logs - that might give you a clue?

> I am wondering if it is timing out on directories containing lots of files, as since it is mounted over the network very large folders might experience latency. I didn't see any relevant settings on the sync command.

rclone doesn't have any timeouts while directory listing.  You say it is mounted over the network.  Is that with NFS?  It might be worth trawling through the NFS logs too.

It might be worth seeing if you can reproduce the problem with find - using an operation that needs to stat each file, say

    find /Users/gryphon/Encrypted -size +1G

See also: possibly related #364 which I never got to the bottom of, but I was convinced it was an encfs problem.

**Comment by gryphonmyers** [comment]
Thanks for the suggestions! I attempted running the transfer again with verbose encfs logging and no errors showed up. Just a bunch of stats. At the time I initially mounted the encfs it did list a few getattr errors about certain meta files and such have filenames that were too short, but that didn't seem relevant. 

It's actually mounted over AFP at the moment. I checked the AFP error log and it didn't contain anything useful. 

As far as I could tell, the find command you suggested completed without any problems. It listed a bunch of paths, then stopped.

The next thing I'm going to try is mounting over SMB instead of AFP. I'd appreciate any other troubleshooting suggestions!
